connector/src/main/scala/com/google/cloud/spark/bigquery/SparkBigQueryOptions.scala

```diff
@@ -270,6 +242,66 @@ object SparkBigQueryOptions {
     getAnyOption(globalOptions, options, name)
       .map(_.toBoolean)
       .getOrElse(defaultValue)
-
 }
 
+  case class IntermediateFormat(dataSource: String,
+                                formatSuffix: String,
+                                formatOptions: FormatOptions)
+  object IntermediateFormat {
+    private val PermittedIntermediateFormats = Map("avro" -> FormatOptions.avro(),
+      "parquet" -> FormatOptions.parquet(),
+      "json" -> FormatOptions.json(),
+      "orc" -> FormatOptions.orc())
+
+    def apply (format: String, sparkVersion: String, sqlConf: SQLConf) : IntermediateFormat = {
+      if (!PermittedIntermediateFormats.contains(format)) {
+        throw new IllegalArgumentException(
+          s"""Data read format '${format}' is not supported.
+             |Supported formats are '${PermittedIntermediateFormats.keys.mkString(",")}'"""
+            .stripMargin.replace('\n', ' '))
+      }
+
+      val formatSuffix = if (format == "json") {
```

Comment:
What's the difference between json and parquet in this case? why not to have `formatSuffix = s".$format"`?

Reply:
We had formatoption.type.tolowercase as the suffix earlier. This does not work for json since type gets resolve to newline_delimited_json.

Reply:
I thought it was the case when the value was of type `FormatOptions`. Now, if I'm not mistaken it is one of the Strings ("avro", "json", "parquet", "orc"), so we can use it directly

Reply:
okay great. I thought there was a techincal reason for getting the type from FormatOptions